src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs

```diff
@@ -53,7 +59,9 @@ public override void ExecuteCmdlet()
         private AppPatch CreateApplicationPatch()
         {
             App existingIotCentralApplication = this.GetApplication();
```

Comment:
We shouldn't be getting the application and setting the properties for the patch. Just send the properties that the user wants to update and let ARM do the resource validation & rp-service do the properties validation.